Applications/Spire/Source/SpireTester/TableModelTester.cpp

```diff
@@ -0,0 +1,250 @@
+#include <boost/variant/get.hpp>
+#include <doctest/doctest.h>
+#include "Spire/Ui/ArrayTableModel.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+
+TEST_SUITE("TableModel") {
+  TEST_CASE("visit_all_operations") {
+    auto model = ArrayTableModel();
```

Comment:
For the unit tests, make use of the minimum number of dependencies needed to ensure correctness, this helps ensure that if something goes wrong, what is wrong is directly related to the specific functionality being tested.

For the visitor, that means removing the use of the ```ArrayTableModel``` and working directly with ```TableModel::Operation```.